src/infrastructure/selectors/XmlSelectorAdapter.spec.js

```diff
@@ -0,0 +1,143 @@
+const { describe, it, beforeEach } = require('mocha');
+const { expect } = require('chai');
+const XmlSelectorAdapter = require('./XmlSelectorAdapter');
+
+describe('XmlSelectorAdapter', () => {
+  let sut;
+
+  beforeEach(() => {
+    sut = new XmlSelectorAdapter();
+  });
+
+  it('should accept xml content type', () => {
+    // given
+    const contentType = 'xml';
+    // when
+    const result = sut.canHandle(contentType);
+    // then
+    expect(result).to.equal(true);
+  });
+
+  it('should accept html content type', () => {
+    // given
+    const contentType = 'html';
+    // when
+    const result = sut.canHandle(contentType);
+    // then
+    expect(result).to.equal(true);
+  });
+
+  it('should return null if no match found', () => {
+    // given
+    const xmlString = '<key1>value1</key1>';
+    const responsePath = '/key2';
+
+    // when
+    const res = sut.select(xmlString, responsePath);
+
+    // then
+    expect(res).to.equal(null, 'Selected data doesn\'t match');
+  });
+
+  it('should select all data when no xpath passed', () => {
+    // given
+    const xmlString = '<key1>value1</key1>';
+    // when
+    const res = sut.select(xmlString, null);
+    // then
+    expect(res).to.equal('<key1>value1</key1>');
+  });
+
+  it('should throw an error when invalid xpath passed', () => {
+    // given
+    const xmlString = '<key1>value1</key1>';
+    const responsePath = '!?X';
+
+    // when, then
+    expect(() => sut.select(xmlString, responsePath)).to.throw();
+  });
+
+  it('should throw an error when invalid xml passed', () => {
+    // given
+    const xmlString = 'value1</key1>';
+    const responsePath = '/key1';
+
+    // when, then
+    expect(() => sut.select(xmlString, responsePath)).to.throw();
+  });
+
+  it('should select simple node from payload', () => {
+    // given
+    const xmlString = '<key1>value1</key1>';
+    const responsePath = '/key1';
+
+    // when
+    const res = sut.select(xmlString, responsePath);
+
+    // then
+    expect(res).to.equal('<key1>value1</key1>', 'Selected data doesn\'t match');
+  });
+
+  it('should select node text from payload', () => {
```

Comment:
I think this (and scenarios below) is redundant - we are testing external xpath and not our code.

Reply:
We at least basic tests can be very helpful. Bug in the external library is still a bug and our soft would not be working correctly.

Reply:
I disagree but well - no harm done by testing more!